### PR TITLE
Fixup calc provided uncertainty

### PR DESCRIPTION
**Pull request by mmore500**
Based on readthrough Friday with @rodsan0

### CODE REVIEW

File: hstrat/_auxiliary_lib/_bit_ceil.py [hunk 1]
```diff
@@ -2,7 +2,8 @@ def bit_ceil(n: int) -> int:
     """Calculate the smallest power of 2 not smaller than n."""
     if n:
         # see https://stackoverflow.com/a/14267825/17332200
-        exp = (n - 1).bit_length()
+        # cast to int to make robust to numpy.int32, numpy.int64, etc.
+        exp = int(n - 1).bit_length()
```
Comment (rodsan0) [inline hstrat/_auxiliary_lib/_bit_ceil.py:6]: Do we need to bounds check for n < 0?

File: hstrat/_auxiliary_lib/_bit_floor.py [hunk 2]
```diff
@@ -5,7 +5,8 @@ def bit_floor(n: int) -> int:
     """
     if n:
         # see https://stackoverflow.com/a/14267825/17332200
-        exp = (n // 2).bit_length()
+        # cast to int to make robust to numpy.int32, numpy.int64, etc.
+        exp = int(n // 2).bit_length()
```
Comment (rodsan0) [inline hstrat/_auxiliary_lib/_bit_floor.py:9]: Same as above

**Review by rodsan0 — APPROVED** [review]
Looks good as-is!

File: hstrat/_auxiliary_lib/_bit_ceil.py (hunk 1 above)
Comment (mmore500) [reply hstrat/_auxiliary_lib/_bit_ceil.py:6]: checked & this definitely doesn't work for negative integers. Adding an assert and a docstring note!

File: hstrat/_auxiliary_lib/_bit_floor.py (hunk 2 above)
Comment (mmore500) [reply hstrat/_auxiliary_lib/_bit_floor.py:9]: ibid

**Pull request merged**